### PR TITLE
Simple UI

### PR DESCRIPTION
**Pull request by CRClothier**
In this checkpoint, I added methods to create books, people and rentals and to display a list of people and books and a list of rentals by person.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -16,26 +16,82 @@ def initialize
...
+    puts 'Has parent permission? [Y/N]:'
+    permission = gets.chomp
+    if permission.upcase == 'Y'
+      permission = true
+    else
+      permission = false
+    end
+    new_person = Student.new(age, '', name, permission)
+    @people << new_person
+    puts 'Person created successfully'
```
Comment (Kingstalux) [inline app.rb]: - [x] Good job so far 👍 When I try to create a student in your console app, I get the following error. Kindly rectify this.
- Error
Annotation 2022-05-19 043628

File: app.rb [hunk 2]
```diff
@@ -16,31 +16,41 @@ def initialize
   end
 
   def list_people
-    @people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"}
+    if @people == []
+      puts
+      puts "No people in system."
+    else
+      @people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"}
+    end
```
Comment (Kingstalux) [inline app.rb]: - [x] After creating the teacher successfully, it is not being displayed. Kindly fix this bug too so as to maintain the functionality of the original design👍
- Your console app:
Annotation 2022-05-19 043713

File: app.rb [hunk 3]
```diff
@@ -16,31 +16,41 @@ def initialize
...
   end
 
   def list_books
-    @books.each { |book| puts "Title: \"#{book.title}\" , Author: #{book.author}"}
+    if @books.length == 0
+      puts
+      puts "No books in system."
+    else
+      @books.each { |book| puts "Title: \"#{book.title}\" , Author: #{book.author}"}
+    end
```
Comment (Kingstalux) [inline app.rb]: - [ ] Same issue here, I can create a book successfully but it is not being displayed. Kindly fix this issue 👍 
- Your console app:
Annotation 2022-05-19 043918

File: app.rb [hunk 4]
```diff
@@ -87,10 +88,12 @@ def create_book
...
   def create_rental
     puts 'Select a book from the following list by number:'
-    @books.each { |book, i| puts "#{i + 1}) Title: \"#{book.title}\" , Author: #{book.author}"}
+    @books.each { |book, i| puts "#{i + 1}) Title: \"#{book.title}\" , Author: #{book.author}" }
     book_index = gets.chomp.to_i - 1
     puts 'Select a person from the following list by number (not id):'
-    @people.each { |person, i| puts "#{i + 1}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"}
+    @people.each do |person, i| 
+      puts "#{i + 1}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
```
Comment (Kingstalux) [inline app.rb]: - [x] Kindly verify this function too as after creating a teacher and a book, it does not display when I try to create a rental.
- [x] Also as I cannot create a rental, I cannot verify the list rental function. Kindly fix your app functionality and make sure everything works as expected 👍

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @CRClothier, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
68747470733a2f2f6d65646961302e67697068792e636f6d2f6d656469612f7741366159414f336a4836526d55494259732f67697068792e676966


## Highlights 👏 
✅ Linters are passing 🔥 
✅ Professional README file 👍 
✅ Good commit history 👏 
✅ Correct Gitflow used 💯 



## Required Changes ♻️
_Check the comments under the review._
- [ ] After making the requested changes, kindly verify that your console app has the correct functionality before submitting for review 😄 
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    puts 'Select a person from the following list by number (not id):'
+    @people.each_with_index do |person, i|
+      puts "#{i + 1}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_index = gets.chomp.to_i - 1
+    print 'Date: '
+    date = gets.chomp.to_i - 1
+    Rental.new(date, @books[book_index], @people[person_index])
+    puts 'Rental created successfully'
+  end
```
Comment (FrederickMih) [inline app.rb]: - [x] Hi @CRClothier, please could you once more address the issues raised by the previous reviewer?. For instance, when creating a student, the app breaks when I was prompted with the option for entering parent permission.


Screenshot from 2022-05-19 06-49-47

Please kindly refer to the previous reviewer comments and screenshots to fix these issues once more. Your student school library app should behave the same way as the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) that was provided to you.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @CRClothier , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Meltrust — APPROVED** [review]
Hi @CRClothier,


## Wow, you did it 🎉
Brilliant
## Thank you for implementing the changes requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### To highlight

- Thank you for correcting and debugging your create person functionality, as requested by the previous reviewer✔️ 

image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**